sequencing-server/src/backgroundTranspiler.ts

```diff
@@ -0,0 +1,230 @@
+    return;
+  }
+
+  // only process 'numberOfThreads' worth at a time ex. transpile 2 logics at a time
+  // This allows for expansion set and sequence expansion to utilize the remaining workers
+  for (let i = 0; i < expansion_rule.length; i += numberOfThreads) {
+    await Promise.all(
+      expansion_rule.slice(i, i + numberOfThreads).map(async expansion => {
+        await promiseThrottler.run(async () => {
+          // Assuming expansion_rule elements have the same type
+          if (expansion instanceof Error) {
+            console.log(`[ Background Transpiler ] Expansion: ${expansion.name} could not be loaded`, expansion);
+            return Promise.reject(`Expansion: ${expansion.name} could not be loaded`);
+          }
+
+          const hash = crypto
+            .createHash('sha256')
+            .update(
+              JSON.stringify({
+                commandDictionaryId,
+                missionModelId,
+                id: expansion.id,
+                expansionLogic: expansion.expansion_logic,
+                activityType: expansion.activity_type,
+              }),
+            )
+            .digest('hex');
+
+          // ignore already transpiled hash info
+          if (typeCheckingCache.has(hash)) {
+            return Promise.resolve();
+          }
+
+          const activitySchema = await activitySchemaDataLoader.load({
+            missionModelId,
+            activityTypeName: expansion.activity_type,
+          });
+
+          // log error
+          if (!activitySchema) {
+            console.log(
+              `[ Background Transpiler ] Activity schema for ${expansion.activity_type} could not be loaded`,
+              activitySchema,
+            );
+            return Promise.reject('Activity schema for ${expansion.activity_type} could not be loaded');
+          }
+
+          const activityTypescript = generateTypescriptForGraphQLActivitySchema(activitySchema);
+
+          // log error
+          if (!activityTypescript) {
+            console.log(
+              `[ Background Transpiler ] Unable to generate typescript for activity ${expansion.activity_type}`,
+              activityTypescript,
+            );
+            return Promise.reject(`Unable to generate typescript for activity ${expansion.activity_type}`);
+          }
+
+          const typecheckingResult = (
+            piscina.run(
+              {
+                expansionLogic: expansion.expansion_logic,
+                commandTypes: commandTypes,
+                activityTypes: activityTypescript,
+                activityTypeName: expansion.activity_type,
+              },
+              { name: 'typecheckExpansion' },
+            ) as ReturnType<typeof typecheckExpansion>
+          ).then(Result.fromJSON);
+
+          typeCheckingCache.set(hash, typecheckingResult);
+          return typecheckingResult;
+        });
+      }),
+    );
+  }
+}
+
+async function fetchLatestMissionModel(): Promise<{
```

Comment:
Can we move all of the GQL function calls out to another file? We have `hasura.ts` with our existing GQL calls right now.

Reply:
Sure, I can do that